lib/fhir/resources/default.js

```diff
@@ -10,30 +10,26 @@
 
 const FhirCommon = require('../common')
 const QueryValidator = require('../query-validator')
+const QueryBuilder = require('../query-builder')
 
 module.exports = (mongo) => {
   const fhirCommon = FhirCommon(mongo)
   const queryValidator = QueryValidator()
+  const queryBuilder = QueryBuilder(mongo)
 
   return {
-    name: 'AuditEvent',
+    name: 'Default',
 
     searchFilters: (ctx, callback) => {
       const queryParams = ctx.query
       const customSupportedParams = {}
 
-      queryValidator.validateQueryParams('AuditEvent', queryParams, customSupportedParams, (badRequest, queryObject) => {
+      queryValidator.validateQueryParams(ctx.resourceType, queryParams, customSupportedParams, (badRequest, queryObject) => {
         if (badRequest) {
           return callback(null, fhirCommon.buildHTTPOutcome(400, 'error', 'invalid', badRequest.message))
         }
 
-        let query = { $and: [] }
-
-        if (queryObject['_id']) {
-          for (const modifier in queryObject['_id']) {
-            query['$and'].push({ 'id': queryObject['_id'][modifier] })
-          }
-        }
+        const query = queryBuilder.buildQuery(ctx.resourceType, queryObject)
```

Comment:
Lets add in the customParams filter here for completeness, There are no custom params at the moment, but possible in future
`const filteredQueryObject = queryUtils.filterOutProperties(queryObject, Object.keys(customSupportedParams))`